src/Parser/Core/Modules/DeathTracker.js

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+
+import { formatMilliseconds } from 'common/format';
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+const debug = true;
+
+class DeathTracker extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  deathTimestamp = 0;
+  resurrectTimestamp = 0;
+  timeDead = 0;
+  castCount = 0;
+  isAlive = true;
+
+  on_toPlayer_death(event) {
+    this.deathTimestamp = this.owner.currentTimestamp;
+    if (debug) {console.log("Player Died @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));}
+    this.isAlive = false;
+  }
+
+  on_toPlayer_resurrect(event) {
+    this.resurrectTimestamp = this.owner.currentTimestamp;
+    this.timeDead += this.resurrectTimestamp - this.deathTimestamp;
+    if (debug) {console.log("Player was Resurrected @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));}
+    this.isAlive = true;
+  }
+
+  on_byPlayer_cast(event) {
+    this.castCount += 1;
+  }
+
+  on_finished() {
+    if (this.isAlive === false) {
+      this.timeDead += this.owner.currentTimestamp - this.deathTimestamp;
```

Comment:
If you want to be really extra, can skip this `on_finished` statement and rename timeDead to `completedTimeDead`, then add 
```
get timeDead() {
   return this.completedTimeDead + (this.isAlive ? 0 : this.owner.currentTimestamp - this.deathTimestamp);
}
```
The advantage of this (other than swag), is that the timeDead value will update correctly as the fight loads, which the user sometimes sees. To be clear, you don't need to update this, I'm mostly just preaching the gospel of the `get` statement.